File: sucraft-keepglass/src/main/java/org/sucraft/keepglass/main/SuCraftKeepGlassPlugin.java
package org.sucraft.keepglass.main;

import org.sucraft.core.common.plugin.PluginUtils;
import org.sucraft.core.common.plugin.SuCraftPlugin;
import org.sucraft.keepglass.listener.GlassBreakListener;

public class SuCraftKeepGlassPlugin extends SuCraftPlugin {
	
	// Singleton
	
	private static SuCraftKeepGlassPlugin instance = null;
	public static SuCraftKeepGlassPlugin instance() {
		return PluginUtils.requirePluginInitialized(instance);
	}
	
	// Initialization
	
	@Override
	public void onPluginPreEnable() {
		instance = this;
	}
	
	// Enable
	
	@Override
	public void onPluginEnable() {
		GlassBreakListener.initialize();
	}
	
}

File: sucraft-keepglass/src/main/kotlin/org/sucraft/keepglass/main/SuCraftKeepGlassPlugin.kt
/*
 * Copyright (c) SuCraft 2021 sucraft.org
 */

package org.sucraft.keepglass.main

import org.sucraft.core.common.sucraft.plugin.SuCraftPlugin
import org.sucraft.core.common.general.pattern.SingletonContainer
import org.sucraft.keepglass.listener.BreakGlassListener


class SuCraftKeepGlassPlugin : SuCraftPlugin() {

	// Companion (singleton)

	companion object : SingletonContainer<SuCraftKeepGlassPlugin>()

	// Enable

	override fun onSuCraftPluginEnable() {
		// Initialize components
		BreakGlassListener
	}

}